### PR TITLE
Development image to test PopPunk branches

### PR DESCRIPTION
**Pull request by absternator**
This PR creates a new dev image that will can be used to deploy specific versions of poppunk. These images have a _-dev postfix. Currently_ to test new poppunk code, it has to be merged and deployed to bioconda. This can take a while slowing us down.

Testing: 
Can test using beebop branch poppunk-version-dev. PR-> pull request https://github.com/bacpop/beebop_py/pull/50.
ALSO deployed to dev currently as well

### CODE REVIEW

**Review by EmmaLRussell — COMMENTED** [review]
When I run this through beebop, by setting API_BRANCH and running the dependencies script, the worker container runs, but the api container crashes on start with the following logs:
```
Skipping virtualenv creation, as specified in config file.
/tmp/tmpqhmv2dg2: line 3:    16 Illegal instruction     (core dumped) poetry run waitress-serve --port=5000 beebop.app:app
ERROR conda.cli.main_run:execute(125): `conda run poetry run waitress-serve --port=5000 beebop.app:app` failed. (See above for error)

```

Also - I'd suggest to only do a dev build when you need to, to minimise build time. 

I feel like this could also use an explanatory note in beebop itself, especially if it's going to mean changing `API_BRANCH` to something which isn't a branch - maybe we should call that `API_IMAGE` or something..

**Comment by absternator** [comment]
Have updated to only produce dev images as needed with flag --with-dev... also moved to ghcr as there were issues with buildkite and images. (mantra found due to different architecture)

**Comment by absternator** [comment]
TODO: remove buildkite agent after testing all works and deployed on dev!!

File: .github/workflows/build_and_push.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+      - uses: actions/checkout@v4
+      - name: Login to GHCR (GitHub Packages)
+        uses: docker/login-action@v3
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+      - name: Build docker image
+        run: ./docker/build --with-dev # TODO: remove --with_dev before merging
+      - name: Push SHA docker image
```
Comment (EmmaLRussell) [inline .github/workflows/build_and_push.yml:24]:
```suggestion
      - name: Push docker image
```
This pushes both branch and tag doesn't it? So this name is a bit misleading.

In some projects we delay pushing the branch tag until tests have passed - can't remember what we were doing in this one on buildkite.

File: README.md [hunk 2]
```diff
@@ -101,3 +101,19 @@ Testing can be done in a second terminal (make sure to activate 'beebop_py') by
 ```
 TESTING=True poetry run pytest
 ```
+
+### Use/Deploy specific version of PopPUNK
```
Comment (EmmaLRussell) [inline README.md:105]:
```suggestion
## Use/Deploy specific version of PopPUNK

```
I think the next two sections are about this top, so this should be a higher level heading.

**Review by EmmaLRussell — APPROVED** [review]
Looks good! Worked for me when I ran from the new registry as beebop dep.
Couple of tiny text suggestions.

**Pull request merged**